plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -234,23 +234,24 @@ public function test_disable_responsive_image_with_picture_element(): void {
 
 		$img_markup = apply_filters( 'the_content', $image );
 
-		// Apply picture element support.
-		$this->opt_in_to_picture_element();
-
-		$picture_markup = apply_filters( 'the_content', $image );
-
 		$img_processor = new WP_HTML_Tag_Processor( $img_markup );
 		$img_processor->next_tag( array( 'tag_name' => 'IMG' ) );
 
-		$this->assertNull( $img_processor->get_attribute( 'sizes' ), 'Sizes attribute missing in IMG tag.' );
+		$this->assertNull( $img_processor->get_attribute( 'sizes' ), 'Make sure that there is no sizes attribute in IMG tag.' );
+		$this->assertNull( $img_processor->get_attribute( 'srcset' ), 'Make sure that there is no srcset attribute in IMG tag.' );
 
-		$picture_processor = new WP_HTML_Tag_Processor( $picture_markup );
-		$picture_processor->next_tag( array( 'tag_name' => 'IMG' ) );
+		// Apply picture element support.
+		$this->opt_in_to_picture_element();
 
-		$this->assertNull( $picture_processor->get_attribute( 'sizes' ), 'Sizes attribute missing in Picture IMG tag.' );
+		$picture_markup = apply_filters( 'the_content', $image );
 
-		$picture_processor->next_tag( array( 'tag_name' => 'source' ) );
+		$picture_processor = new WP_HTML_Tag_Processor( $picture_markup );
+		$this->assertFalse( $img_processor->next_tag( array( 'tag_name' => 'picture' ) ), 'Make sure that there is no Picture tag.' );
+		$this->assertFalse( $img_processor->next_tag( array( 'tag_name' => 'source' ) ), 'Make sure that there is no source tag.' );
+		$this->assertTrue( $picture_processor->next_tag( array( 'tag_name' => 'IMG' ) ), 'Make sure that there is a IMG tag.' );
+		$this->assertNull( $picture_processor->get_attribute( 'sizes' ), 'Make sure that there is no sizes attribute in IMG tag.' );
+		$this->assertNull( $picture_processor->get_attribute( 'srcset' ), 'Make sure that there is no srcset attribute in IMG tag.' );
 
-		$this->assertNull( $picture_processor->get_attribute( 'sizes' ), 'Sizes attribute missing in Picture source tag.' );
+		$this->assertSame( $img_markup, $picture_markup, 'Make sure both markup are the same' );
```

Comment:
I don't believe this is correct per https://github.com/WordPress/performance/issues/1448#issuecomment-2276202488.

When no `srcset` is available, then the `PICTURE > SOURCE` tag should have its `srcset` attribute populated with the `src` for the modern image format. When a JPEG fallback is available, we still need to use a `PICTURE` element so that browsers that don't support the modern image format will get served the JPEG instead. 

As seen in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture#the_type_attribute), there's no need for `srcset` to have any width or pixel descriptors:

```html
<picture>
  <source srcset="photo.avif" type="image/avif" />
  <source srcset="photo.webp" type="image/webp" />
  <img src="photo.jpg" alt="photo" />
</picture>
```

So think of `srcset` as a superset of what `src` can contain.

Reply:
Addressed in https://github.com/WordPress/performance/pull/1449/commits/7e09a7231dc04ef3772c0be58fae185dc89a3b7b